universum/modules/artifact_collector.py

```diff
@@ -173,24 +173,21 @@ def set_and_clean_artifacts(self, project_configs: Configuration, ignore_existin
             with self.structure.block(block_name=name, pass_errors=True):
                 self.preprocess_artifact_list(artifact_list, ignore_existing_artifacts)
 
-    def get_conditional_step_branches_artifacts(self, step: Step) -> List[ArtifactInfo]:
-        artifacts: List[Optional[ArtifactInfo]] = [
-            self.get_config_artifact_if_exists(step.if_succeeded),
-            self.get_config_artifact_if_exists(step.if_failed),
-            self.get_config_artifact_if_exists(step.if_succeeded, is_report_artifact=True),
-            self.get_config_artifact_if_exists(step.if_failed, is_report_artifact=True)
-        ]
+    def get_conditional_step_branches_artifacts(self, conditional_step: Step) -> List[ArtifactInfo]:
+        steps_to_process: List[Step] = []
+        if conditional_step.if_succeeded:
+            steps_to_process.extend(list(conditional_step.if_succeeded.all()))
```

Comment:
Please also fix Configuration.filter function to recursively handle `if_succeeded` and `if_failed` members.
Without such fix, these members will remain unfiltered.

Reply:
This is a known issue and is recorded in the conditional steps [task](https://github.com/Samsung/Universum/issues/709) as a separate work item. As I remember from our past discussions, there are several options for implementing conditional steps filtering and it should be properly designed and described in the documentation beforehand.
So I prefer to postpone the steps filtering fix to a moment when it will be prepared as a complete solution.